### PR TITLE
DOC: organize content in advanced guide

### PR DESCRIPTION
**Pull request by jakevdp**
cc @8bitmp3, @canyon289  

Preview: https://jax--14158.org.readthedocs.build/en/14158/advanced_guide.html

### CODE REVIEW

File: docs/advanced_guide.rst [hunk 1]
```diff
@@ -2,20 +2,41 @@
 
 Advanced Tutorials
 ==================
+This section contains examples and tutorials on more advanced topics
```
Comment (canyon289) [inline docs/advanced_guide.rst:5]: A short explanation of what makes these advanced would be nice, even for beginner users. That way they have a sense what advanced usage means and if they even need to be advanced

Here's a suggestion

```suggestion
This section contains examples and tutorials on more advanced topics, such as Multi Core computation, Custom operations, and more in depth applications
```

File: docs/advanced_guide.rst [hunk 2]
```diff
@@ -2,20 +2,41 @@
...
 .. toctree::
+   :caption: Examples
+   :maxdepth: 1
+
+   notebooks/neural_network_with_tfds_data
+   notebooks/Neural_Network_and_Data_Loading
+   notebooks/vmapped_log_probs
+
+.. toctree::
+   :caption: Multi-core and Multi-pod Computation
```
Comment (yashk2810) [inline docs/advanced_guide.rst:16]: Is POD an external name? i.e. something we use in cloud TPU docs?

Comment (jakevdp) [reply docs/advanced_guide.rst:16]: I've seen it used elsewhere (e.g. https://blogs.nvidia.com/blog/2021/03/05/what-is-a-cluster-pod/). Do you think multi-cluster or something like that would be more clear?

Comment (jakevdp) [reply docs/advanced_guide.rst:16]: Maybe "parallel and distributed computation"?

Comment (yashk2810) [reply docs/advanced_guide.rst:16]: I think we should ask Skye or James

Comment (jakevdp) [reply docs/advanced_guide.rst:16]: James recommended "Parallel Computation"

**Review by yashk2810 — APPROVED** [review]

**Pull request merged**